packages/website/docs/components/patterns/AI Writing Assistant/main.js

```diff
@@ -0,0 +1,162 @@
+import '@ui5/webcomponents/dist/Label.js';
+import '@ui5/webcomponents/dist/Input.js';
+import "@ui5/webcomponents/dist/Menu.js";
+import "@ui5/webcomponents/dist/MenuItem.js";
+import "@ui5/webcomponents-icons/dist/ai.js";
+import "@ui5/webcomponents-icons/dist/stop.js";
+import "@ui5/webcomponents-icons/dist/navigation-left-arrow.js";
+import "@ui5/webcomponents-icons/dist/navigation-right-arrow.js";
+
+import announce from "@ui5/webcomponents-base/dist/util/InvisibleMessage.js";
+import InvisibleMessageMode from "@ui5/webcomponents-base/dist/types/InvisibleMessageMode.js";
+
+const input = document.getElementById('ai-wa-input');
+const icon = document.getElementById('ai-wa-icon');
+const menuGenerate = document.getElementById('ai-wa-menu-generate');
+const menuRegenerate = document.getElementById('ai-wa-menu-regenerate');
+
+let lastInputValue = "";
+let loadTimeout;
+
+
+input.addEventListener('focus', () => {
+	updateIconVisibility(true);
+});
+
+input.addEventListener('focusout', e => {
+	if (!menuGenerate.open && !menuRegenerate.open) {
+		updateIconState("ai", "AI Writing Assistant (Shift + F4)");
+		updateIconVisibility(false);
+	}
+});
+
+icon.addEventListener("mousedown", e => {
+	if (icon.name === "stop") {
```

Comment:
mousedown probably not the best event. it triggers the logic for action before user make confirmation with releasing the mouse key.